metrics/web/test/base/presentation/graphs/animated_bar_test.dart

```diff
@@ -0,0 +1,242 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:metrics/base/presentation/graphs/animated_bar.dart';
+import 'package:metrics/base/presentation/widgets/rive_animation.dart';
+import 'package:rive/rive.dart';
+
+import '../../../test_utils/testbed/rive_animation_testbed.dart';
+
+// ignore_for_file: avoid_redundant_argument_values
+
+void main() {
+  group("AnimatedBar", () {
+    const asset = 'asset';
+    final riveAnimationFinder = find.byType(RiveAnimation);
+
+    testWidgets(
+      "throws an AssertionError if the given height is null",
+      (WidgetTester tester) async {
+        await tester.pumpWidget(
+          const _AnimatedBarTestbed(
+            height: null,
+          ),
+        );
+
+        expect(tester.takeException(), isAssertionError);
+      },
+    );
+
+    testWidgets(
+      "throws an AssertionError if the given height is negative",
+      (WidgetTester tester) async {
+        await tester.pumpWidget(
+          const _AnimatedBarTestbed(
+            height: -1.0,
+          ),
+        );
+
+        expect(tester.takeException(), isAssertionError);
+      },
+    );
+
+    testWidgets(
+      "throws an AssertionError if the given height equals to zero",
+      (WidgetTester tester) async {
+        await tester.pumpWidget(
+          const _AnimatedBarTestbed(
+            height: 0.0,
+          ),
+        );
+
+        expect(tester.takeException(), isAssertionError);
+      },
+    );
+
+    testWidgets(
+      "throws an AssertionError if the given width is null",
+      (WidgetTester tester) async {
+        await tester.pumpWidget(
+          const _AnimatedBarTestbed(
+            width: null,
+          ),
+        );
+
+        expect(tester.takeException(), isAssertionError);
+      },
+    );
+
+    testWidgets(
+      "throws an AssertionError if the given width is negative",
+      (WidgetTester tester) async {
+        await tester.pumpWidget(
+          const _AnimatedBarTestbed(
+            width: -1.0,
+          ),
+        );
+
+        expect(tester.takeException(), isAssertionError);
+      },
+    );
+
+    testWidgets(
+      "throws an AssertionError if the given width equals to zero",
+      (WidgetTester tester) async {
+        await tester.pumpWidget(
+          const _AnimatedBarTestbed(
+            width: 0.0,
+          ),
+        );
+
+        expect(tester.takeException(), isAssertionError);
+      },
+    );
+
+    testWidgets(
+      "applies the given height to the animated bar",
+      (WidgetTester tester) async {
+        const expectedHeight = 20.0;
+
+        await tester.pumpWidget(
+          const _AnimatedBarTestbed(
+            height: expectedHeight,
+          ),
+        );
+
+        final animatedBar = tester.widget<AnimatedBar>(find.byType(AnimatedBar));
```

Comment:
It seems like we should take a `SizedBox` laying above the `RiveAnimation` widget and check if the `AnimatedBar` widget applies the given `height` and `width` to this widget.